music_assistant/providers/audible_audiobooks/__init__.py

```diff
@@ -0,0 +1,284 @@
+            self.helper = AudibleHelper(
+                mass=self.mass,
+                client=self._client,
+                provider_instance=self.instance_id,
+                provider_domain=self.domain,
+            )
+            self.logger.info("Successfully authenticated with Audible.")
+        except Exception as e:
+            self.logger.error(f"Failed to authenticate with Audible: {e}")
+            raise LoginFailed("Failed to authenticate with Audible.")
+
+    @property
+    def supported_features(self) -> set[ProviderFeature]:
+        """Return the features supported by this Provider."""
+        return (ProviderFeature.BROWSE, ProviderFeature.LIBRARY_AUDIOBOOKS)
+
+    @property
+    def is_streaming_provider(self) -> bool:
+        """Return True if the provider is a streaming provider."""
+        return True
+
+    async def get_library_audiobooks(self) -> AsyncGenerator[Audiobook, None]:
+        """Get all audiobooks from the library."""
+        # Convert to async generator by using async yield
+        async for audiobook in self.helper.get_library():
+            yield audiobook
+
+    async def get_audiobook(self, prov_audiobook_id: str) -> Audiobook:
+        """Get full audiobook details by id."""
+        return await self.helper.get_audiobook(asin=prov_audiobook_id, use_cache=False)
+
+    async def get_stream_details(
+        self, item_id: str, media_type: MediaType = MediaType.TRACK
+    ) -> StreamDetails:
+        """Get streamdetails for a audiobook based of asin."""
+        return await self.helper.get_stream(asin=item_id)
+
+    async def on_streamed(
+        self, streamdetails: StreamDetails, seconds_streamed: int, fully_played: bool
+    ) -> None:
+        """Handle callback when an item completed streaming."""
+        await self.helper.set_last_position(streamdetails.item_id, seconds_streamed)
+
+    async def unload(self, is_removed: bool = False) -> None:
+        """
+        Handle unload/close of the provider.
+
+        Called when provider is deregistered (e.g. MA exiting or config reloading).
+        is_removed will be set to True when the provider is removed from the configuration.
+        """
+        if is_removed:
+            self.helper.deregister()
+
+    async def resolve_image(self, path: str) -> str | bytes:
+        """
+        Resolve an image from an image path.
+
+        This either returns (a generator to get) raw bytes of the image or
+        a string with an http(s) URL or local path that is accessible from the server.
+        """
+        return path
+
+    async def browse(self, path: str) -> Sequence[MediaItemType | ItemMapping]:
+        """Browse this provider's items.
+
+        :param path: The path to browse, (e.g. provider_id://artists).
+        """
+        audiobooks = []
+        async for audiobook in self.get_library_audiobooks():
+            audiobooks.append(audiobook)
+        return audiobooks
```

Comment:
If you intend to just returning the library books, you can leave this out as that is the default behavior